static/src/javascripts/projects/common/modules/navigation/user-account.js

```diff
@@ -2,110 +2,54 @@
 
 import fastdom from 'fastdom';
 import { getUserFromCookie, isUserLoggedIn } from 'common/modules/identity/api';
-import avatarAPI from 'common/modules/avatar/api';
 
 const updateCommentLink = (): void => {
-    const commentItem = document.querySelector('.js-show-comment-activity');
-    const commentLink =
-        commentItem &&
-        commentItem.querySelector('.js-add-comment-activity-link');
-
-    if (commentItem && commentLink) {
-        const user = getUserFromCookie();
-        if (user) {
-            fastdom.write(() => {
-                commentItem.classList.remove('u-h');
-                commentLink.setAttribute(
-                    'href',
-                    `https://profile.theguardian.com/user/id/${user.id}`
-                );
-            });
-        }
-    }
-};
-
-const enhanceAvatar = (): Promise<void> => {
-    const fallbackEl = document.querySelector(
-        '.js-navigation-account-avatar-fallback'
-    );
-    const avatarEl = document.querySelector('.js-navigation-account-avatar');
-
-    const preloadAvatar = (src: string): Promise<void> => {
-        const image = new Image();
-        image.src = src;
-
-        return new Promise(resolve => {
-            image.onload = resolve;
-        });
-    };
-
-    const swapFallback = (
-        fallback: ?HTMLElement,
-        avatar: ?HTMLElement,
-        src: string
-    ) => {
-        fastdom.write(() => {
-            if (fallback) {
-                fallback.classList.add('u-h');
-            }
-
-            if (avatar) {
-                avatar.setAttribute('src', src);
-                avatar.classList.remove('u-h');
+    fastdom
+        .read(() => [...document.querySelectorAll('.js-show-comment-activity')])
+        .then(commentItems => {
+            const user = getUserFromCookie();
+
+            if (user) {
+                commentItems.forEach(commentItem => {
+                    const commentLink = commentItem.querySelector(
+                        '.js-add-comment-activity-link'
+                    );
+
+                    if (commentLink) {
```

Comment:
Maybe:
```js
fastdom.read(() => commentItem.querySelector('.js-add-comment-activity-link'))
  then((commentItem) => {
    if (commentItem) {
      // ...
    }
  });
```

Reply:
Hmm... Yes I agree. Why don't I try to refactor this today/tomorrow instead?